packages/e2e-tests/plugins/delete-installed-fonts.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * Plugin Name: Gutenberg Test Delete Installed Fonts
+ * Plugin URI: https://github.com/WordPress/gutenberg
+ * Author: Gutenberg Team
+ *
+ * @package gutenberg-test-delete-installed-fonts
+ */
+
+/**
+ * Delete all user installed fonts, associated font files, the fonts directory, and user global styles typography
+ * setings for the current theme so that we can test uploading/installing fonts in a clean environment.
+ */
+function gutenberg_delete_installed_fonts() {
+	$font_family_ids = new WP_Query(
+		array(
+			'post_type'      => 'wp_font_family',
+			'posts_per_page' => -1,
+			'fields'         => 'ids',
+		)
+	);
+
+	// Delete all font families, their child font faces, and associated font files.
+	foreach ( $font_family_ids->posts as $font_family_id ) {
+		wp_delete_post( $font_family_id, true );
+	}
+
+	// Delete the font directory, which should now be empty.
+	$font_path = wp_get_font_dir()['path'];
+
+	if ( is_dir( $font_path ) ) {
+		rmdir( $font_path );
+	}
```

Comment:
In some development environments the content and uploads folder is shared between the test suite and the developer's manual testing so deleting the directory will fail as it will contain files.

If the files are deleted beforehand then the manual testing environment will be busted.

For the test suite I suggest filtering the folder and cleaning it up afterwards. You can probably use `wp_generate_uuid4()` to generate the folder name.

Reply:
Good point, that part is brittle. I've modified the plugin to use a randomly generated directory for fonts during the duration of the test and to cleanup on plugin activation and deactivation. Let me know what you think!

Reply:
Thanks Grant, that looks good. Is there an e2e tear down at the end that could be used to delete the temporary folder? It's no big deal if there isn't.

Reply:
I think that because of the way we're testing here with a plugin to handle the server-side logic, the tear-down happens on deactivation on the plugin, i.e. on [line 19](https://github.com/WordPress/gutenberg/pull/60221/files#diff-64c2c3d6d0f3e7e9b5025b67f4dcd9377fa04fd10b49e9553362a30aeb06e02eR19) we deactivate the plugin after all tests:

```
test.afterAll( async ( { requestUtils } ) => {
	await requestUtils.deactivatePlugin(
		'gutenberg-test-delete-installed-fonts'
	);
} );
```

And the plugin handles the deletion of fonts and folders on deactivation:

```
register_deactivation_hook( __FILE__, 'gutenberg_delete_installed_fonts' );
```